js/userscript.js

```diff
@@ -1,51 +1,253 @@
+          user.signedIn = true;
+        }
+      }
+      return user;
+    });
+    localStorage.setItem('users', JSON.stringify(users));
+  }
+
+  if (signedInUserTrue === true) {
+    userPasswordInput.value = '';
+    userPageSetup();
+  } else {
+    userPasswordInput.value = '';
+    signInInvalid.style.display = 'block';
+    signInForm.classList.add('active', 'show');
+  }
+}
+
+function userPageSetup() {
+  let signUpDiv = document.getElementById('signUpDiv');
+
+  if (!localStorage.getItem("users")) {
+    signUpDiv.style.display = 'block';
+    signUpSignIn();
+  }
+  if (JSON.parse(localStorage.getItem('users'))) {
+    userDetails = JSON.parse(localStorage.getItem('users'));
+    let signedInUser = userDetails.filter((user) => user.signedIn === true);
+    console.log(signedInUser)
+
+    if (signedInUser.length !== 0) {
+      signUpDiv.style.display = 'none';
+        userName = signedInUser[0].userName;
+        heading.innerText = `Welcome, ${userName}`;
+        document.getElementById('logOutIcon').addEventListener('click', () => {
+          console.log('im clicked');
+          signOut();
+        });
+        userPage();
+    } else {
+      signUpSignIn();
+      }
+    } 
+  }
+
+
+function signOut() {
+  let userDetails = JSON.parse(localStorage.getItem('users'));
+  let usersPageContainer = document.getElementById('usersPageContainer');
+  let heading = document.getElementById('heading');
+  let headingInnerText = document.getElementById('heading').innerText;
+  let userNameInHeading = headingInnerText.replace(`Welcome, `, "");
+  let users = userDetails.map((user) => {
+    if (user.userName === userNameInHeading){
+      console.log(userNameInHeading)
+      user.signedIn = false;
+    }
+    return user;
+  });
+  console.log(users)
+  localStorage.setItem('users', JSON.stringify(users));
+  heading.innerText = `Welcome`;
+  usersPageContainer.style.display = 'none';
+  userPageSetup();
 }
 
-function userPage() {
-    let userDetails = JSON.parse(localStorage.getItem('user'));
+// userPage();
+function userPage() {  
+  let usersPageContainer = document.getElementById('usersPageContainer');
+  usersPageContainer.style.display = 'block';
+  let usersPage = document.getElementById('usersPage');
+  let headingInfo = document.getElementById('headingInfo');
+  headingInfo.style.display = 'none';
+  let favShowsHeading = document.getElementById('favShowsHeading');
+  favShowsHeading.innerText = 'Favourite Shows';
+  let userDetails = JSON.parse(localStorage.getItem('users'));
+
+  let signedInUser = userDetails.filter((user) => user.signedIn === true);
+  console.log(signedInUser);
+
+
+  if (!signedInUser[0].favShows) {
+    let noFavShowText = `<h4>Click the TV icon to search for and add your favourite shows</h4>`;
+    return usersPage.innerHTML = noFavShowText;
+  }
+  let favShows = signedInUser[0].favShows;
+  let shows = getAllShows();
+  let favShowsList = shows.filter((show) => {
```

Comment:
A `filter` function should return `true` or false, whereas you're returning a `string` value (the `show` variable). Your function still works because non-empty strings are [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) in Javascript, so `return show;` has the same effect as `return true;` here. But a clearer way to write the filter would be:

```js
let favShowsList = shows.filter((show) => {
    for (let i = 0; i < favShows.length; i++) {
        return (show.id == favShows[i]);
    }
  });
```

And if you want to be extra-precise and change your `==` to `===`, you can cast both variables to numbers using the `+` operator:

```js
return (+show.id === +favShows[i]);
```